crates/aptos/src/stake/mod.rs

```diff
@@ -193,16 +195,51 @@ pub struct WithdrawStake {
 }
 
 #[async_trait]
-impl CliCommand<TransactionSummary> for WithdrawStake {
+impl CliCommand<Vec<TransactionSummary>> for WithdrawStake {
     fn command_name(&self) -> &'static str {
         "WithdrawStake"
     }
 
-    async fn execute(mut self) -> CliTypedResult<TransactionSummary> {
-        self.node_op_options
-            .submit_transaction(aptos_stdlib::stake_withdraw(self.amount))
-            .await
-            .map(|inner| inner.into())
+    async fn execute(mut self) -> CliTypedResult<Vec<TransactionSummary>> {
+        let client = self
+            .node_op_options
+            .rest_options
+            .client(&self.node_op_options.profile_options)?;
+        let amount = self.amount;
+        let owner_address = self.node_op_options.sender_address()?;
+        let mut transaction_summaries: Vec<TransactionSummary> = vec![];
+
+        let stake_pool_results = get_stake_pools(&client, owner_address).await?;
+        for stake_pool in stake_pool_results {
+            match stake_pool.pool_type {
+                StakePoolType::Direct => {
+                    transaction_summaries.push(
+                        self.node_op_options
+                            .submit_transaction(aptos_stdlib::stake_withdraw(amount))
+                            .await
+                            .map(|inner| inner.into())?,
+                    );
+                },
+                StakePoolType::StakingContract => {
+                    transaction_summaries.push(
```

Comment:
This is going to withdraw more than requested.  What I suggest is you add a select for which pool unless there's a way to prevent multiple pools for an account.

As a stopgap this should be fine but please follow up with the select which pool (if more than one)

Reply:
This can be a follow up change IMO as all of the owner accounts currently in both testnet and mainnet only have one stake pool.